aiida_codtools/cli/misc/cif_import.py

```diff
@@ -0,0 +1,142 @@
+    """
+    from CifFile.StarFile import StarError
+    from urllib2 import HTTPError
+    from aiida.tools.dbimporters import DbImporterFactory
+
+    importer_parameters = {}
+    query_parameters = {}
+
+    if importer_server is not None:
+        importer_parameters['server'] = importer_server
+
+    if importer_db_host is not None:
+        importer_parameters['host'] = importer_db_host
+
+    if importer_db_name is not None:
+        importer_parameters['db'] = importer_db_name
+
+    if importer_db_password is not None:
+        importer_parameters['passwd'] = importer_db_password
+
+    if importer_api_url is not None:
+        importer_parameters['url'] = importer_api_url
+
+    if importer_api_key is not None:
+        importer_parameters['api_key'] = importer_api_key
+
+    if number_species is not None:
+
+        if database == 'mpds':
+
+            query_parameters = {
+                'query': {},
+                'collection': 'structures'
+            }
+
+            if number_species == 1:
+                query_parameters['query']['classes'] = 'unary'
+            elif number_species == 2:
+                query_parameters['query']['classes'] = 'binary'
+            elif number_species == 3:
+                query_parameters['query']['classes'] = 'ternary'
+            else:
+                raise ValueError('only unaries, binaries and ternaries are supported for {} database'.format(database))
+
+        else:
+            query_parameters['number_of_elements'] = number_species
+
+    importer_class = DbImporterFactory(database)
+    importer = importer_class(**importer_parameters)
+
+    try:
+        query_results = importer.query(**query_parameters)
+    except BaseException as exception:
+        click.echo('database query failed: {}'.format(exception))
+        return
+
+    existing_cif_nodes = [cif.get_attr('source')['id'] for cif in group.nodes]
+
+    counter = 0
+    for entry in query_results:
+
+        source_id = entry.source['id']
+
+        if source_id in existing_cif_nodes:
+            click.echo('Cif with source id <{}> already present in group {}'.format(source_id, group.name))
+            continue
+
+        try:
+            cif = entry.get_cif_node()
+        except (StarError, HTTPError) as exception:
+            click.echo('Encountered an error retrieving cif data: {}'.format(exception))
+        else:
+            try:
+                if cif.has_partial_occupancies():
+                    click.echo('Cif with source id <{}> has partial occupancies, skipping'.format(source_id))
+                    continue
+                else:
+                    cif.store()
+
+                    group.add_nodes([cif])
+
+                    counter += 1
+                    click.echo('Newly stored CifData<{}> with source id <{}>'.format(cif.pk, source_id))
+            except ValueError:
+                click.echo('Cif with source id <{}> has partial occupancies, skipping'.format(source_id))
```

Comment:
it seems the message here was copied from the one in line 127, and is not meaningful.
The old version of the import script was saying:
```
                # has_partial_occupancies() raises ValueError if occupancy
                # can not be converted to float value (usually it means
                # that there are occupancy values '?' or '.' -- in some
                # cases these values belong to dummy atoms). Such
                # structures are rare and can be skipped as incomplete so
                # as not to bias the research).

```
so I presume we can say something like
`click.echo('Occupancies of cif with source id <{}> cannot be converted to floats, skipping'.format(source_id))`